runtime/src/bank.rs

```diff
@@ -4811,17 +4811,20 @@ impl Bank {
         TransactionBatch::new(lock_results, self, Cow::Borrowed(transactions))
     }
 
-    /// Prepare a transaction batch without locking accounts for transaction simulation.
-    pub(crate) fn prepare_simulation_batch(
-        &self,
-        transaction: SanitizedTransaction,
+    /// Prepare a transaction batch from a single transaction without locking accounts
+    pub(crate) fn prepare_unlocked_batch_from_single_tx<'a>(
+        &'a self,
+        transaction: &'a SanitizedTransaction,
     ) -> TransactionBatch<'_, '_> {
         let tx_account_lock_limit = self.get_transaction_account_lock_limit();
         let lock_result = transaction
             .get_account_locks(tx_account_lock_limit)
             .map(|_| ());
-        let mut batch =
-            TransactionBatch::new(vec![lock_result], self, Cow::Owned(vec![transaction]));
+        let mut batch = TransactionBatch::new(
+            vec![lock_result],
+            self,
+            Cow::Borrowed(std::slice::from_ref(transaction)),
```

Comment:
*minor*

I think in most cases, elements are imported, up to a certain meaningful parent, rather than referenced with the full path.  Even for `std`:

```rust
use std::slice;
```

And then
```suggestion
            Cow::Borrowed(slice::from_ref(transaction)),
```

Reply:
nice! done 32871055205dc0afa11170e3c459afdc8deae04c